src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -12,11 +14,34 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public T save(T figure) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(figure);
+            transaction.commit();
+            return figure;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert figure " + figure, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Figure WHERE color = :color", clazz)
+                    .setParameter("color", color)
+                    .getResultList();
```

Comment:
I think you can get class cast exception if your query  return unexpected figure (for example you can get red circle and cast it to triangle).In my opinion we must use here criteriaquery and get only figures where class is clazz